### PR TITLE
[HERA-453] [LMS] upgrades hera_pages xblock rendering and adds table handling

### PR DESCRIPTION
**Pull request by Golub-Sergey**
**Description:** upgrades hera_pages xblock rendering and adds table handling

**Youtrack:** [ticket](https://youtrack.raccoongang.com/issue/HERA-453)

### CODE REVIEW

File: hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html [hunk 1]
```diff
@@ -2,22 +2,94 @@
...
+                            % if item.get('tableData'):
+                                <div class="table-block">
+                                    <% user_answer_input_index = 0 %>
+                                    <form class="table-form-${index}">
+                                        <table>
+                                            <tbody>
+                                                % if item['tableData'].get('columns'):
+                                                    <tr>
+                                                        % for column in item['tableData']['columns']:
+                                                            % if column['type'] == 'head':
```
Comment (sendr) [inline hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html]:
```suggestion
                                                            % if column.get('type') == 'head':
```

File: hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html [hunk 2]
```diff
@@ -2,22 +2,94 @@
...
+                                                                        <%
+                                                                            try:
+                                                                                row_answer = user_answers[index][user_answer_input_index] if user_answers else ''
+                                                                            except IndexError:
+                                                                                row_answer = ''
+                                                                        %>
+                                                                        <input type="text" name="${index}" value="${row_answer}"/>
+                                                                    </td>
+                                                                    <% user_answer_input_index += 1 %>
+                                                                % elif row[str(idx)]['type'] == 'head':
```
Comment (sendr) [inline hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html]:
```suggestion
                                                                % elif row[str(idx)].get('type') == 'head':
```

Comment (sendr) [reply hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html]: not all cells have this key (as appeared, need to be fixed in cms part)

File: hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html (hunk 1 above)
Comment (sendr) [reply hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html]: not all cells have this key (as appeared, need to be fixed in cms part)

File: hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html (hunk 2 above)
Comment (Golub-Sergey) [reply hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html]: Done

File: hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html (hunk 1 above)
Comment (Golub-Sergey) [reply hera_xblocks/hera_pages/hera_pages/static/html/hera_pages.html]: Done

File: hera_xblocks/hera_pages/hera_pages/static/js/src/hera_pages.js [hunk 3]
```diff
@@ -45,4 +59,44 @@ function HeraPagesXBlock(runtime, element) {
             // asNavFor: '.slider-nav'
         });
     });
+
+    $questionForm.on('change blur keyup', function(e) {
```
Comment (sendr) [inline hera_xblocks/hera_pages/hera_pages/static/js/src/hera_pages.js]: will check all this file in the next PR

**Pull request merged**